test/functional/fixtures/quarantine-mode/index-test.js

```diff
@@ -1,96 +1,51 @@
-var expect         = require('chai').expect;
-var http           = require('http');
-var promisifyEvent = require('promisify-event');
-var config         = require('../../config');
+var expect                = require('chai').expect;
+var config                = require('../../config');
+var quarantineModeTracker = require('../../quarantine-mode-tracker');
 
 
-var TEST_SERVER_PORT               = 3002;
-var TEST_SERVER_KEEP_ALIVE_TIMEOUT = 2000;
-var BROWSERS_COUNT                 = config.browsers.length;
+function checkQuarantineTestRuns (quarantineState) {
+    var browsers = quarantineState === 'failed-quarantine' ?
+                   quarantineModeTracker.browsersFailingQuarantine :
+                   quarantineModeTracker.browsersPassingQuarantine;
 
+    var browserAgents = Object.keys(browsers);
 
-describe('Quarantine mode tests', function () {
-    var expectedBrowsersCount = BROWSERS_COUNT;
-    var requestNumber         = 0;
-    var responses             = [];
-    var serverResponse        = null;
-
-    function requestHandler (request, response) {
-        responses.push(response);
-        expectedBrowsersCount--;
-
-        if (expectedBrowsersCount > 0)
-            return;
-
-        responses.forEach(function (res) {
-            res.writeHead(200, {
-                'Access-Control-Allow-Origin': '*'
-            });
-
-            res.write(serverResponse(requestNumber));
-
-            res.end();
-        });
-
-        expectedBrowsersCount = BROWSERS_COUNT;
-        responses             = [];
-        requestNumber++;
-    }
-
-    var testServer = null;
-
-    function startServer () {
-        testServer = http.createServer(requestHandler);
-
-        testServer.setTimeout(TEST_SERVER_KEEP_ALIVE_TIMEOUT);
-        testServer.listen(TEST_SERVER_PORT);
-        return promisifyEvent(testServer, 'listening');
-    }
+    expect(browserAgents.length).to.equal(config.browsers.length);
 
-    function closeServer () {
-        testServer.close();
-        return promisifyEvent(testServer, 'close');
-    }
-
-    before(function () {
-        return startServer();
+    // NOTE: Last step is 4th:
```

Comment:
hmmm... you are talking about how many times a test is launched in a quarantine, don't you?
so maybe they are not *steps* but rather *attempts* or *runs*?